skyfield/elementslib.py

```diff
@@ -9,17 +9,28 @@
                    sinh, sqrt, arccos, arctanh, zeros_like, ones_like, divide,
                    where, pi, cross)
 
-def osculating_elements_of(position, reference_frame=None):
+def osculating_elements_of(position, reference_frame=None, gm_km3_s2=None):
     """Produce the osculating orbital elements for a position.
 
-    The ``position`` should be an :class:`~skyfield.positionlib.ICRF`
-    instance like that returned by the ``at()`` method of any Solar
-    System body, specifying a position, a velocity, and a time.  An
-    instance of :class:`~skyfield.elementslib.OsculatingElements` is
-    returned.
+    Parameters:
+    position : :class:`~skyfield.positionlib.ICRF` object
+         commonly returned by the ``at()`` method of any Solar System body
+    reference_frame: 3x3 numpy array, optional
+        commonly used values are found in skyfield.data.spice.inertial_frames
+    gm_km3_s2: float, optional
+        Gravitational parameter (G*M) in units of km^3/s^2 If not specified, this is calculated for you.
 
+    Returns:
+    :class:`~skyfield.elementslib.OsculatingElements`
     """
-    mu = GM_dict.get(position.center, 0) + GM_dict.get(position.target, 0)
+    if not gm_km3_s2:
+        gm_km3_s2 = GM_dict.get(position.center, 0)
+        if position.center not in range(0, 10): # true if position.center is not a barycenter
```

Comment:
Building a `range()` object each time sounds expensive. Maybe could could build a `frozenset()` of the numbers 0 through 9 and use an `in` test against that? (Other possibilities are possible, like an `isinstance` against `int` followed by a normal inequality.)

Oh — and, instead of having a comment, try splitting this into two lines, the first of which does the test and assigns the true/false to a name like `is_barycenter`. Then no comment is needed for readers to understand the meaning.